prov/efa/src/rxr/rxr_pkt_type_data.c

```diff
@@ -50,10 +50,10 @@ int rxr_pkt_init_data(struct rxr_ep *ep,
 	data_hdr->type = RXR_DATA_PKT;
 	data_hdr->version = RXR_PROTOCOL_VERSION;
 	data_hdr->flags = 0;
-	data_hdr->recv_id = tx_entry->rx_id;
+	data_hdr->recv_id = rx_entry->tx_id;
```

Comment:
I do not think this is right. It should be something like
```
if (op_entry->type == RXR_TX_ENTRY)
    data_hdr->recv_id = op_entry->tx_id;
else
   data_hdr->recv_id = op_entry->rx_id;
```

Reply:
That's a very good catch

Reply:
That's a very good catch. I think we need to adapt this also in rxr_pkt_init_cts().  Am I correct?